Sources/ScipioKit/Producer/PIF/PIFGenerator.swift

```diff
@@ -53,89 +71,302 @@ struct PIFGenerator {
+                        umbrella header "\(headerPath.moduleEscapedPathString)"
+                        export *
+                    }
+                """
+            case .umbrellaDirectory(let directoryPath):
+                settings[.MODULEMAP_FILE_CONTENTS] = """
+                    framework module \(c99Name) {
+                        umbrella "\(directoryPath.moduleEscapedPathString)"
+                        export *
+                    }
+                """
+            case .none:
+                settings[.MODULEMAP_FILE_CONTENTS] = nil
+            }
+        } else {
+            let bridgingHeaderName = settings[.SWIFT_OBJC_INTERFACE_HEADER_NAME] ?? "\(name)-Swift.h"
+            settings[.MODULEMAP_FILE_CONTENTS] = """
                 framework module \(c99Name) {
-                    header "\(name)-Swift.h"
+                    header "\(bridgingHeaderName)"
                     export *
                 }
-                """
-                        }
+            """
+        }
 
-                        // If the built framework is named same as one of the target in the package, it can be picked up
-                        // automatically during indexing since the build system always adds a -F flag to the built products dir.
-                        // To avoid this problem, we build all package frameworks in a subdirectory.
-                        settings[.BUILT_PRODUCTS_DIR] = "$(BUILT_PRODUCTS_DIR)/PackageFrameworks"
-                        settings[.TARGET_BUILD_DIR] = "$(TARGET_BUILD_DIR)/PackageFrameworks"
+        configuration.buildSettings = settings
+        return configuration
+    }
 
-                        configuration.buildSettings = settings
-                        return configuration
-                    }
+    private func collectPublicHeaders(of clangTarget: ClangTarget) -> Set<AbsolutePath> {
+        let publicHeaders = clangTarget
+            .headers
+            .filter { $0.isDescendant(of: clangTarget.includeDir) }
+        let notSymlinks = publicHeaders.filter { !fileSystem.isSymlink($0) }
+        let symlinks = publicHeaders.filter { fileSystem.isSymlink($0) }
 
-                    pifTarget.buildConfigurations = newConfigurations
-                }
-            }
+        // Sometimes, public headers include a file and its symlink both.
+        // This situation raises a duplication error
+        // So duplicated symlinks have to be omitted
+        let notDuplicatedSymlinks = symlinks.filter { path in
+            notSymlinks.allSatisfy { FileManager.default.contentsEqual(atPath: path.pathString, andPath: $0.pathString) }
         }
-        return pif
+
+        return Set(notSymlinks + notDuplicatedSymlinks)
+    }
+
+    private func guid(_ suffixes: String...) -> String {
+        "GUID::SCIPIO::\(pifTarget.name)::" + suffixes.joined(separator: "::")
+    }
+
+    private func addPublicHeaders(clangTarget: ClangTarget) {
```

Comment:
The current `PIFBuilder` of SwiftPM doesn't add public headers to `HeadersBuildPhase`. So we have to add public headers manually.